CONTRIBUTING.md

```diff
@@ -0,0 +1,98 @@
+
+## Contribute code
+
+Commit style is detailed in the [osrd.fr/docs](https://osrd.fr/en/docs/guides/contribute/code/).
+
+### Pull request / code review
+
+- Every PR, good or bad, is an act of generosity.
+  Opening with a positive comment will help the author feel rewarded, and your subsequent remarks may be heard more clearly.
+  You may feel good also.
+- Add a description to PRs to explain what they do and why.
+- Each PR must be linked to at least one github issue to simplify tracking.
+- Pay attention to all the comments made by reviewers and either answer/challenge them or address them before asking for another review.
+- Assign one maintainer and one developer using github assignment when opening the PR to get reviews.
+- Help the reviewer by following advice given in [mtlynch article](https://mtlynch.io/code-review-love/).
+- Review code following advice given in [osrd.fr/docs](https://osrd.fr/en/docs/guides/contribute/code-reviews/) and [mtlynch article](https://mtlynch.io/human-code-reviews-1/).
+- Make commits of logical and atomic units.
+- If possible, make PR of logical and atomic units too (avoid mixing refactoring, new features and bug fix at the same time).
+- Keep your branch up-to-date :
+```
+git checkout <your_branch>
+git fetch
+git rebase origin/dev
+```
+
+### Bugs
+
+The way support is delivered greatly impacts how users perceive the product and the team.
+Handling the incoming support requests efficiently is thus a top priority.
+
+- Bug must have a correct description and the bug's issue template must be filled carefully.
+- Bug must be tagged with:
+  - `kind:bug`
+  - one or several `area:<affected_area>`
+  - one `severity:<bug_severity>` if possible, if severity is not known leave it blank it will be added later by another team member
+- Contributors can change issues' tags (severity, area, kind, ...) if necessary. 
+  Keep in mind that explaining why you are editing an issue is the best way to avoid misunderstanding.
+- If you are working on a bug or plan to work on a bug, assign yourself to the bug.
+  If a bug is assigned to someone, consider that they are is already working on the resolution.
+- PRs solving bugs should add a regression tests to ensure that bug will not be back in the future.
+
+### General code recommendations
+
+- Take the time to pick good names.
+  Avoid non well-known abbreviations.
+
+### Python
+
+- Follow [the Zen of Python](https://www.python.org/dev/peps/pep-0020/).
+- To keep our API surface minimal, only document and make public what is necessary.
+- Code is linted with [flake8](https://github.com/csachs/pyproject-flake8).
+- Code is formatted with [Black](https://github.com/psf/black).
+- Imports are sorted with [Isort](https://github.com/PyCQA/isort).
+
+### Test
+
+- TODO
+
+### Rust
+
+- To keep our API surface minimal, only document and make public what is necessary.
```

Comment:
I get keeping stuff private, but shouldn't we document as much as possible anyway?

Reply:
Not sure, for instance java code is polluted by doc like `/** Constructor */` that is not helping but just adding content. Lots of the doc is just paraphrasing methods/functions/classes names and I think more doc doesn't mean better doc. If naming is good and goal is obvious then there is no need to add a paraphrasing doc that maybe won't be updated in the future.

Good documentation must help future us to understand what is done not just paraphrasing or documenting obvious and simple code.

Reply:
Ah yes, that kind of documentation is useless. But that's true for public APIs as well, I'm not sure why you would draw a distinction.

Reply:
We should probably have specific guideline on the doc part (not necessarily in this iteration of the Contributing)

While what is necessary to make public is usually quite obvious (the compiler enforces it after all - even if designing god APIs takes work), what is necessary to document is both non obvious and quite often subject to debate.

Maybe (as a short first pass)

```
- To keep our API surface minimal, only make public what is necessary.
- Documentation should only be there to explain things that aren't self evident.
```